src/plugins/smith_waterman/src/SmithWatermanAlgorithm.cpp

```diff
@@ -174,8 +174,9 @@ void SmithWatermanAlgorithm::calculateMatrixForMultipleAlignmentResult() {
     n = pat_n * 2;
     unsigned int dirn = (4 + pat_n + 3) >> 2;
     unsigned int memory = n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn;
-    int *buf, *matrix = (int *)malloc(memory);
-    if (matrix == nullptr) {
+    quint64 quint_memory = (quint64)n * sizeof(int) + (quint64)pat_n * (quint64)0x80 + (quint64)matrixLength * (quint64)dirn;
+    int* buf, * matrix = (int*)malloc(memory);
```

Comment:
Does malloc accept 64-bit value?
I would suggest to mark algorithm as failed if required memory size is > some 32-bit constant like 1Gb.

Reply:
![image](https://user-images.githubusercontent.com/17490668/134481531-0bbf173d-6d68-44fd-a3e2-1fc396a179f6.png)
![image](https://user-images.githubusercontent.com/17490668/134482602-4e2af36f-f2ae-48fb-a450-f39d30d7de5b.png)
As i can see on my MSVC size_t is unsigned int64. So it should be acceptable.

I don't think this is good idea to add any constant limits to allocating values. If malloc couldn't allocate desireble amount of memory following code will handle this situation. In that way we got flexible algorithm which can work on systems with different RAM volume.

Reply:
if you expect that malloc works correctly with 64bit values and this scale is good for SW algorithm why do you have the  'quint_memory != memory' check below?